Titanium.Web.Proxy/Network/Tcp/TcpConnectionFactory.cs

```diff
@@ -46,12 +46,17 @@ internal class TcpConnectionFactory
             TcpClient client;
             Stream stream;
 
+            bool isLocalhost = IsLocalhost(remoteHostName);

```

Comment:
DNS resolution can be slow at times. Since we need this flag only when external proxy is set, I would like to check first for external proxy and then for this flag.